platform/contracts/admin/src/endpoints.rs

```diff
@@ -134,14 +137,23 @@ pub fn sudo(deps: DepsMut<'_>, env: Env, msg: SudoMsg) -> ContractResult<CwRespo
             admin_contract,
             migration_spec,
             post_migration_execute,
-        }) => crate::contracts::migrate(
-            deps.storage,
-            env.contract.address,
-            release,
-            admin_contract,
-            migration_spec,
-            post_migration_execute,
-        )
+        }) => if let Some(admin_contract) = admin_contract {
```

Comment:
Let's not combine two scenarios under the same API. That makes it difficult to use.

Define a dedicated `SudoMsg::MigrateSelf` ... or even better, let's use the native migration for the Admin. Then remove the ability to migrate the Admin with a SudoMsg.